docs/frontend/webpack-source-maps.md

```diff
@@ -6,8 +6,402 @@ which allows a developer to view / debug developer-friendly source code instead
+- [Source Maps Primer](#source-maps-primer)
+- [Integrating Source Maps](#integrating-source-maps)
+  - [Open Source Apps](#open-source-apps)
+  - [Private Apps](#private-apps)
+- [Source Map Example](#source-map-example)
+- [Advantages](#advantages)
+- [Disadvantages](#disadvantages)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+##### Basic Example
+
+(Example source available at: [github.com/FormidableLabs/formidable-playbook/tree/master/examples/frontend/src/es5](https://github.com/FormidableLabs/formidable-playbook/tree/master/examples/frontend/src/es5))
+
+Let's start with the familiar source code used in our
+[shared library](./webpack-shared-libs.md)) and
+[code splitting example](./webpack-source-maps.md)) examples.
+
+
+[`foo.js`](../../examples/frontend/src/es5/foo.js)
+
+```js
+module.exports = function (id, msg) {
+  return "<h1 id=\"" + id + "\">" + msg + "</h1>";
+};
+```
+
+[`app1.js`](../../examples/frontend/src/es5/app1.js)
+
+```js
+var foo = require("./foo");
+
+document.querySelector("#content").innerHTML += foo("app1", "App 1");
+```
+
+[`app2.js`](../../examples/frontend/src/es5/app2.js)
+
+```js
+var foo = require("./foo");
+
+document.querySelector("#content").innerHTML += foo("app2", "App 2");
+```
+
+##### Source Maps Primer
+
+Modern JavaScript on the frontend practically _guarantees_ that the original
+source code developers write is different from the code distributed and
+executed on web browsers. These days, code is likely to be transformed by things
+like:
+
+* Babel transforms of ESnext/JSX code to ES5 code.
+* Developer-friendly full code to minified gibberish for smaller download size
+  and some protection of application IP.
+
+The skew between what developers write and what a browser executes makes
+debugging JavaScript applications quite tedious and complicated. Fortunately,
+modern browsers implement **source maps** whereby minified code can be
+mapped to real developer sources before execution to give developers a much
+better debugging experience.
+
+If you are not familiar with how source maps work, please take a moment to
+review a couple introductory articles:
+
+* http://www.html5rocks.com/en/tutorials/developertools/sourcemaps/
+* https://github.com/ryanseddon/source-map/wiki/Source-maps:-languages,-tools-and-other-info
+* https://blog.sentry.io/2015/10/29/debuggable-javascript-with-source-maps.html
+
+The brief summary of how source maps work is as follows:
+
+* During bundling, an application is configured to output a source mapping file
+  corresponding to the application bundle.
+* The real application bundle is downloaded and parsed by a browser.
+* If the browser encounters a special control comment like:
+  `//# sourceMappingURL=PATH/TO/bundle.js.map` **and** the developer console
+  is open, then the source map file is downloaded from the URL in the control
+  comment.
+* Once downloaded, the mapped file is swapped in and debuggable in place of the
```

Comment:
Things I'm wondering at this point (that may or may not be helpful)...
What does "swapped in" mean? Is the mapped file actually _executing_ now? or is it just shown in the dev console? What if the page and scripts were loaded and _then_ the dev console was opened? Would we have to refresh?